### ISSUE TITLE
Directories field causes Yarn crash

### ISSUE THREAD

**Issue by daniluk4000** [lindell/JsBarcode#438]
Hello there. Directories field of this specific library causes Yarn bug https://github.com/yarnpkg/berry/issues/5766 to happen.

[Removal of directories](https://github.com/Vages/JsBarcode/commit/18bccf61cab82a0991b4092632c2da7f5b087bdc) solves this problem. I guess if directories are need to be used, they can be replaced with exports field.

**Comment by daniluk4000** [comment]
Workaround: use @Vages fork of library:

```json
{
  "dependencies": {
    "jsbarcode": "Vages/JsBarcode#30eb5a989e70adc23383c8faf48634640a308bd4",
  }
}
```

**Comment by daniluk4000** [comment]
@lindell can see you are still active, not sure if you watch issues though, can you please fix this so we don't have to use fork?

**Comment by lindell** [comment]
@daniluk4000  Hi. I keep a watch for critical issues and make sure those does not exist, but got burnt out on doing other things a while back.

I'm happy to fix this if it is actually a problem in JsBarcode, but it seems to be a yarn bug, that is shown for jsbarcode and some other projects?

**Comment by daniluk4000** [comment]
> I'm happy to fix this if it is actually a problem in JsBarcode, but it seems to be a yarn bug, that is shown for jsbarcode and some other projects?

I thought so as well at first time, but then I've noticed that out of ~1000 dependencies jsbarcode is the only one that breaking yarn. Same in yarn issue - it was originally created with jsbarcode case.

**Comment by lindell** [comment]
After looking at bin. It seems that it is used wrongly in package.json. Will remove it.

**Comment by Vages** [comment]
I am charmed that you recommend my fork, @daniluk4000. Be aware that people I know have experienced https://github.com/yarnpkg/berry/issues/3500 when using it (I have added the following warning to the repository description):

> Be aware that you may still experience https://github.com/yarnpkg/berry/issues/3500, and I can‘t do anything about that.

So long!

**Comment by daniluk4000** [comment]
> After looking at bin. It seems that it is used wrongly in package.json. Will remove it.

Fixed in 3.11.6. Works like a charm for me in latest Yarn v3 & Yarn v4. Closing this. Thank you!

> I am charmed that you recommend my fork

You are welcome! Your issue and fork helped to actually determine the problem and remove upgrade blocker. Before your issue, I wasn't even sure what package causes this and why. Thanks for that!
